app/qml/map/RecordingTools.qml

```diff
@@ -182,29 +181,33 @@ Item {
 
     recordingMapTool: mapTool
 
-    onGpsSwitchClicked: {
+    function centerToGpsAndFollow()
+    {
       if ( root.gpsState.state === "unavailable" ) {
         showMessage( qsTr( "GPS currently unavailable." ) )
         return
       }
 
+
       mapTool.centeredToGPS = true
       root.map.mapSettings.setCenter( mapPositioning.mapPosition )
+      __appSettings.autoCenterMapChecked = true;
+    }
+
+    onGpsSwitchClicked: {
+      centerToGpsAndFollow()
     }
 
     onGpsSwithHeld: {
+      centerToGpsAndFollow()
+
       // start / stop streaming mode
       if ( mapTool.recordingType === RecordingMapTool.Manual )
       {
         mapTool.recordingType = RecordingMapTool.StreamMode
 
         // add first point immediately
         mapTool.addPoint( crosshair.recordPoint )
-        root.map.mapSettings.setCenter( mapPositioning.mapPosition )
-      }
-      else
-      {
-        mapTool.recordingType = RecordingMapTool.Manual
```

Comment:
Let's keep the `else` branch here so that it is possible to exit the streaming mode